compiler/hash-parser/src/gen.rs

```diff
@@ -1214,40 +2390,47 @@ where
             let (op, consumed_tokens) = Operator::from_token_stream(self);
 
             match op {
-                None => break,
-                Some(op) => {
+                // check if the operator here is re-assignable, as in '+=', '/=', if so then we need to stop
+                // parsing onwards because this might be an assignable expression...
+                // Only perform this check if know prior that the expression is not made of compounded components.
+                Some(op) if !op.assignable => {
                     // consume the number of tokens eaten whilst getting the operator...
-                    (0..consumed_tokens).for_each(|_| {
-                        self.next_token();
-                    });
+                    self.offset.update(|x| x + consumed_tokens as usize);
```

Comment:
Why is consumed_tokens not a usize already?

Reply:
It's a `u8` because the number of chars can't be larger than a `u8`.